Update LDC to 1.41.0 and adjust LLVM/Clang compatibility

corneliusroemer opened the pull request:
- Updated LDC version to 1.41.0.
- Updated SHA256s for Unix and Windows sources.
- Adjusted LLVM and Clang compatibility to support up to LLVM 20.

corneliusroemer committed:
Update LDC to 1.41.0 and adjust LLVM/Clang compatibility.

- Updated LDC version to 1.41.0.
- Updated SHA256s for Unix and Windows sources.
- Adjusted LLVM and Clang compatibility to support up to LLVM 20.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/meta.yaml**:

 * ℹ️ The recipe is not parsable by parser `conda-souschef (grayskull)`. This parser is not currently used by conda-forge, but may be in the future. We are collecting information to see which recipes are compatible with grayskull.
 * ℹ️ License is not an SPDX identifier (or a custom LicenseRef) nor an SPDX license expression.

Documentation on acceptable licenses can be found [here]( https://conda-forge.org/docs/maintainer/adding_pkgs.html#spdx-identifiers-and-expressions ).


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/15937524971](https://github.com/conda-forge/conda-forge-webservices/actions/runs/15937524971). Examine the logs at this URL for more detail.</sub>

corneliusroemer commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 25.5.0, conda-smithy 3.50.1, and conda-forge-pinning 2025.06.27.21.37.34

corneliusroemer committed:
Pin ldc version to 1.26.0 for linux-64 bootstrapping.

This addresses the 'undefined reference to getrandom' error by ensuring glibc compatibility.
